router/core/router.go

```diff
@@ -119,7 +119,7 @@ type (
 		configPoller configpoller.ConfigPoller
 		selfRegister selfregister.SelfRegister
 
-		registrationInfo *nodev1.RegistrationInfo
+		publicKey *ecdsa.PublicKey
```

Comment:
`registrationInfo` was only ever used to create the `publicKey`. We eagerly create the key instead on the first `router.Start`.